ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/StackComponentView.kt

```diff
@@ -68,58 +79,119 @@ internal fun StackComponentView(
+        if (style.badge != null) {
+            when (style.badge.style) {
+                Badge.Style.Overlay -> {
+                    StackWithOverlaidBadge(style, state, style.badge.stackStyle, style.badge.alignment, modifier)
+                }
 
-        // Modifier irrespective of dimension.
-        val commonModifier = remember(style) {
-            Modifier
-                .padding(style.margin)
-                .applyIfNotNull(shadowStyle) { shadow(it, style.shape) }
-                .applyIfNotNull(backgroundColorStyle) { background(it, style.shape) }
-                .clip(style.shape)
-                .applyIfNotNull(borderStyle) { border(it, style.shape) }
-                .padding(style.padding)
-        }
+                Badge.Style.EdgeToEdge -> {
+                }
 
-        val content: @Composable () -> Unit = remember(style.children) {
-            @Composable { style.children.forEach { child -> ComponentView(style = child, state = state) } }
+                Badge.Style.Nested -> {
+                }
+            }
+        } else {
+            MainStackComponent(style, state, modifier)
         }
+    }
+}
 
-        // Show the right container composable depending on the dimension.
-        when (style.dimension) {
-            is Dimension.Horizontal -> Row(
-                modifier = modifier
-                    .size(style.size, verticalAlignment = style.dimension.alignment.toAlignment())
-                    .then(commonModifier),
-                verticalAlignment = style.dimension.alignment.toAlignment(),
-                horizontalArrangement = style.dimension.distribution.toHorizontalArrangement(
-                    spacing = style.spacing,
-                ),
-            ) { content() }
+@Composable
+private fun StackWithOverlaidBadge(
+    style: StackComponentStyle,
+    state: PaywallState.Loaded.Components,
+    badgeStack: StackComponentStyle,
+    alignment: TwoDimensionalAlignment,
+    modifier: Modifier = Modifier,
+) {
+    var badgeHeight by remember {
+        mutableIntStateOf(0)
+    }
+    val paddingDp = with(LocalDensity.current) { (badgeHeight / 2f).toDp() }
+    val padding = when (alignment) {
+        TwoDimensionalAlignment.TOP_LEADING,
+        TwoDimensionalAlignment.TOP,
+        TwoDimensionalAlignment.TOP_TRAILING,
+        -> PaddingValues(top = paddingDp)
+        TwoDimensionalAlignment.BOTTOM_LEADING,
+        TwoDimensionalAlignment.BOTTOM,
+        TwoDimensionalAlignment.BOTTOM_TRAILING,
+        -> PaddingValues(bottom = paddingDp)
+        else -> PaddingValues(0.dp)
+    }
+    Box(modifier = modifier) {
+        // TODO Fix margins when using badges
+        MainStackComponent(style, state, modifier = Modifier.padding(padding))
+        StackComponentView(
+            badgeStack,
+            state,
+            modifier = Modifier
+                .align(alignment.toAlignment())
+                .onGloballyPositioned {
```

Comment:
I don't really like using this, since emerge/snapshot tests usually don't pick these changes correctly... Not sure if there is a better way to do this, while allowing arbitrary contents inside the badge though...